syntax = "proto3";

message Response{
  repeated Message messages = 1;
  string cursor = 2;
  int64 fetchInterval = 3;
  int64 now = 4;
  string internalExt = 5;
  int32 fetchType = 6;
  map<string, string> routeParams = 7;
  int64 heartbeatDuration = 8;
  bool needAck = 9;
  string pushServer = 10;
  message Message{
    string method = 1;
    bytes payload = 2;
    int64 msgId = 3;
    int32 msgType = 4;
    int64 offset = 5;
  }
}

message ChatMessage {
  Common common = 1;
  User user = 2;
  string content = 3;
  bool visibleToSender = 4;
  Image backgroundImage = 5;
  string fullScreenTextColor = 6;
  Image backgroundImageV2 = 7;
  Image giftImage = 10;
}

message GiftMessage {
  Common common = 1;
  int64 giftId = 2;
  int64 fanTicketCount = 3;
  int64 groupCount = 4;
  int64 repeatCount = 5;
  int64 comboCount = 6;
  User user = 7;
  User toUser = 8;
  int32 repeatEnd = 9;
  int64 groupId = 11;
  int64 incomeTaskgifts = 12;
  int64 roomFanTicketCount = 13;
  GiftStruct gift = 15;
  string logId = 16;
  int64 sendType = 17;
  int64 bannedDisplayEffects = 20;
}

message Common {
  string method = 1;
  int64 msgId = 2;
  int64 roomId = 3;
  int64 createTime = 4;
  int32 monitor = 5;
  bool isShowMsg = 6;
  string describe = 7;
  int64 foldType = 9;
  int64 anchorFoldType = 10;
  int64 priorityScore = 11;
  string logId = 12;
  string msgProcessFilterK = 13;
  string msgProcessFilterV = 14;
  User user = 15;
  int64 anchorFoldTypeV2 = 17;
  int64 processAtSeiTimeMs = 18;
}

message User {
  int64 id = 1;
  int64 shortId = 2;
  string nickname = 3;
  int32 gender = 4;
  string signature = 5;
  int32 level = 6;
  int64 birthday = 7;
  string telephone = 8;
  Image avatarThumb = 9;
  Image avatarMedium = 10;
  Image avatarLarge = 11;
  bool verified = 12;
  int32 experience = 13;
  string city = 14;
  int32 status = 15;
  int64 createTime = 16;
  int64 modifyTime = 17;
  int32 secret = 18;
  string shareQrcodeUri = 19;
  int32 incomeSharePercent = 20;
  Image badgeImageList = 21;
}

message Image {
  repeated string urlList = 1;
  string uri = 2;
  int64 height = 3;
  int64 width = 4;
  string avgColor = 5;
  int32 imageType = 6;
  string openWebUrl = 7;
  //webcast content = 8;
  bool isAnimated = 9;
}

message GiftStruct {
  Image image = 1;
  string describe = 2;
  bool notify = 3;
  int64 duration = 4;
  int64 id = 5;
  //webcast fansclubInfo = 6;
  bool forLinkmic = 7;
  bool doodle = 8;
  bool forFansclub = 9;
  bool combo = 10;
  int32 type = 11;
  int32 diamondCount = 12;
  bool isDisplayedOnPanel = 13;
  int64 primaryEffectId = 14;
  Image giftLabelIcon = 15;
  string name = 16;
  string region = 17;
  string manual = 18;
  bool forCustom = 19;
  Image icon = 21;
  int32 actionType = 22;
  int32 watermelonSeeds = 23;
  string goldEffect = 24;
  int64 goldenBeans = 26;
  int64 honorLevel = 27;
  int32 itemType = 28;
  string schemeUrl = 29;
  //webcast giftOperation = 30;
  string eventName = 31;
  int64 nobleLevel = 32;
  string guideUrl = 33;
  bool punishMedicine = 34;
  bool forPortal = 35;
  string businessText = 36;
  bool cnyGift = 37;
  int64 appId = 38;
  int64 vipLevel = 39;
  bool isGray = 40;
  string graySchemeUrl = 41;
  int64 giftScene = 42;
  //webcast giftBanner = 43;
  bool forFirstRecharge = 46;
  Image dynamicImgForSelected = 47;
  int32 afterSendAction = 48;
  int64 giftOfflineTime = 49;
  string topBarText = 50;
  Image topRightAvatar = 51;
  string bannerSchemeUrl = 52;
  bool isLocked = 53;
  int64 reqExtraType = 54;
  //webcast giftPreviewInfo = 56;
  //webcast giftTip = 57;
  int32 needSweepLightCount = 58;
}